src/schema/me/order/update_order_mutation.js

```diff
@@ -0,0 +1,91 @@
+import { GraphQLInputObjectType, GraphQLNonNull, GraphQLString } from "graphql"
+import { OrderType } from "schema/me/order"
+import { mutationWithClientMutationId } from "graphql-relay"
+
+export const ShippingAddressInputType = new GraphQLInputObjectType({
+  name: "ShippingAddressInput",
+  fields: {
+    name: {
+      type: GraphQLString,
+    },
+    street: {
+      type: GraphQLString,
+    },
+    city: {
+      type: GraphQLString,
+    },
+    region: {
+      type: GraphQLString,
+    },
+    postal_code: {
+      type: GraphQLString,
+    },
+    country: {
+      type: GraphQLString,
+    },
+    use_id: {
+      type: GraphQLString,
+    },
+  },
+})
+
+export const OrderInputType = new GraphQLInputObjectType({
+  name: "OrderInput",
+  fields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "Order to be updated",
+    },
+    email: {
+      type: GraphQLString,
+      description: "Email address, required of an unauthenticated user",
+    },
+    notes: {
+      type: GraphQLString,
+      description: "Additional notes",
+    },
+    session_id: {
+      type: GraphQLString,
+      description: "Session ID necessary if there is no user present",
+    },
+    shipping_address: {
+      type: ShippingAddressInputType,
+    },
+    telephone: {
+      type: GraphQLString,
+      description: "Buyer's telephone",
+    },
+  },
+})
+
+export default mutationWithClientMutationId({
+  name: "UpdateOrder",
+  decription: "Update an order",
+  inputFields: OrderInputType.getFields(),
+  outputFields: {
+    order: {
+      type: OrderType,
+      resolve: order => order,
+    },
+  },
+  mutateAndGetPayload: (
+    {
+      id,
+      email,
+      notes,
+      session_id,
+      shipping_address: { name, street, city, region, postal_code, use_id },
+      telephone,
+    },
+    request,
+    { rootValue: { updateOrderLoader } }
+  ) => {
+    return updateOrderLoader(id, {
```

Comment:
You might do a check for the required `email`/`session_id` fields. `if (!accessToken) { /* raise error if missing */`, otherwise Gravity will just (presumably) throw that error.

Reply:
I like that! Since it can make the logic more explicit here.

In general, though, how much of the logic in gravity's endpoints do we want to replicate here? (thinking about future mutations). I guess this becomes less relevant when we're stitching in the endpoints from their sources.